### ISSUE TITLE
[Bug] Fail to run dataset in parallel mode

### ISSUE THREAD

**Issue by wdmhouston** [GEOSX/GEOSX#1448]
**Describe the bug**
[test.zip](https://github.com/GEOSX/GEOSX/files/6563871/test.zip)
When run test dataset in parallel mode, error reported(no error in non-parallel mode):
````
user@ubuntu:/data/final1_rate$ mpirun -np 10 geosx -i CO2.xml
Adding Solver of type CompositionalMultiphaseReservoir, named coupledFlowAndWells
Adding Solver of type CompositionalMultiphaseFVM, named compositionalMultiphaseFlow
Adding Solver of type CompositionalMultiphaseWell, named compositionalMultiphaseWell
Adding Mesh: InternalMesh, mesh
Adding Mesh: InternalWell, wellInjector1
Adding Event: PeriodicEvent, outputs
Adding Event: PeriodicEvent, restarts
Adding Event: PeriodicEvent, timeHistoryCollection
Adding Event: PeriodicEvent, timeHistoryOutput
Adding Event: PeriodicEvent, wellRateCollection
Adding Event: PeriodicEvent, timeRateHistoryOutput
Adding Event: PeriodicEvent, solverApplications
Adding Output: VTK, vtkOutput
Adding Output: Restart, restartOutput
Adding Output: TimeHistory, timeHistoryOutput
Adding Output: TimeHistory, timeRateHistoryOutput
   TableFunction: permxFunc
   TableFunction: permyFunc
   TableFunction: permzFunc
Adding Object CellElementRegion named Reservoir from ObjectManager::Catalog.
Adding Object WellElementRegion named wellRegion from ObjectManager::Catalog.
   TableFunction: CO2DensityTable
   TableFunction: CO2ViscosityTable
   TableFunction: brineDensityTable
   TableFunction: CO2SolubilityTable
***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/mpiCommunications/SpatialPartition.cpp:204
***** Controlling expression (should be false): check != m_size
***** Rank 0:
Expected check == m_size
  check = 1
  m_size = 10


** StackTrace of 7 frames **
Frame 0: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 1: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 2: geosx::ProblemManager::generateMesh()
Frame 3: geosx::ProblemManager::problemSetup()
Frame 4: geosx::GeosxState::initializeDataRepository()
Frame 5: main
Frame 6: __libc_start_main
Frame 7: _start
=====

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
Received signal 15: Terminated

** StackTrace of 20 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 2: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 3: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 4: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 5: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 6: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 7: __cxa_demangle
Frame 8: LvArray::system::demangle[abi:cxx11](char const*)
Frame 9: geosx::dataRepository::Wrapper<LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer> >& geosx::dataRepository::Group::registerWrapper<LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer>*)
Frame 10: geosx::ObjectManagerBase::ObjectManagerBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 11: geosx::ElementSubRegionBase::ElementSubRegionBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 12: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 13: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 14: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 15: geosx::ProblemManager::generateMesh()
Frame 16: geosx::ProblemManager::problemSetup()
Frame 17: geosx::GeosxState::initializeDataRepository()
Frame 18: main
Frame 19: __libc_start_main
Frame 20: _start
=====

Received signal 15: Terminated

** StackTrace of 23 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 2: std::__cxx11::basic_stringbuf<char, std::char_traits<char>, std::allocator<char> >::overflow(int)
Frame 3: std::basic_streambuf<char, std::char_traits<char> >::xsputn(char const*, long)
Frame 4: std::basic_ostream<char, std::char_traits<char> >& std::__ostream_insert<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, char const*, long)
Frame 5: conduit::Schema::path[abi:cxx11]() const
Frame 6: conduit::Schema::path[abi:cxx11]() const
Frame 7: conduit::Schema::path[abi:cxx11]() const
Frame 8: conduit::Schema::path[abi:cxx11]() const
Frame 9: conduit::Schema::path[abi:cxx11]() const
Frame 10: conduit::Schema::path[abi:cxx11]() const
Frame 11: conduit::Node::path[abi:cxx11]() const
Frame 12: geosx::dataRepository::Wrapper<LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer> >& geosx::dataRepository::Group::registerWrapper<LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LvArray::Array<int, 1, camp::int_seq<long, 0l>, long, LvArray::ChaiBuffer>*)
Frame 13: geosx::ObjectManagerBase::ObjectManagerBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 14: geosx::ElementSubRegionBase::ElementSubRegionBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 15: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 16: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 17: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 18: geosx::ProblemManager::generateMesh()
Frame 19: geosx::ProblemManager::problemSetup()
Frame 20: geosx::GeosxState::initializeDataRepository()
Frame 21: main
Frame 22: __libc_start_main
Frame 23: _start
=====

Received signal 15: Terminated

** StackTrace of 13 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 2: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 3: __cxa_demangle
Frame 4: LvArray::system::demangle[abi:cxx11](char const*)
Frame 5: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 6: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 7: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 8: geosx::ProblemManager::generateMesh()
Frame 9: geosx::ProblemManager::problemSetup()
Frame 10: geosx::GeosxState::initializeDataRepository()
Frame 11: main
Frame 12: __libc_start_main
Frame 13: _start
=====

Received signal 15: Terminated

** StackTrace of 16 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: conduit::DataType::compatible(conduit::DataType const&) const
Frame 2: conduit::Node::init(conduit::DataType const&)
Frame 3: conduit::Node::fetch(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
Frame 4: geosx::dataRepository::WrapperBase::WrapperBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group&)
Frame 5: geosx::dataRepository::Wrapper<LvArray::SortedArray<long, long, LvArray::ChaiBuffer> >& geosx::dataRepository::Group::registerWrapper<LvArray::SortedArray<long, long, LvArray::ChaiBuffer>, LvArray::SortedArray<long, long, LvArray::ChaiBuffer> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long*)
Frame 6: geosx::ObjectManagerBase::ObjectManagerBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 7: geosx::ElementSubRegionBase::ElementSubRegionBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 8: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 9: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 10: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 11: geosx::ProblemManager::generateMesh()
Frame 12: geosx::ProblemManager::problemSetup()
Frame 13: geosx::GeosxState::initializeDataRepository()
Frame 14: main
Frame 15: __libc_start_main
Frame 16: _start
=====

Received signal 15: Terminated

** StackTrace of 21 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: __dynamic_cast
Frame 2: bool std::has_facet<std::ctype<char> >(std::locale const&)
Frame 3: std::basic_ios<char, std::char_traits<char> >::_M_cache_locale(std::locale const&)
Frame 4: std::basic_ios<char, std::char_traits<char> >::init(std::basic_streambuf<char, std::char_traits<char> >*)
Frame 5: conduit::Schema::name[abi:cxx11]() const
Frame 6: conduit::Schema::path[abi:cxx11]() const
Frame 7: conduit::Schema::path[abi:cxx11]() const
Frame 8: conduit::Schema::path[abi:cxx11]() const
Frame 9: conduit::Schema::path[abi:cxx11]() const
Frame 10: conduit::Schema::path[abi:cxx11]() const
Frame 11: conduit::Node::path[abi:cxx11]() const
Frame 12: geosx::dataRepository::Wrapper<geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> > >& geosx::dataRepository::Group::registerWrapper<geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> >*)
Frame 13: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 14: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 15: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 16: geosx::ProblemManager::generateMesh()
Frame 17: geosx::ProblemManager::problemSetup()
Frame 18: geosx::GeosxState::initializeDataRepository()
Frame 19: main
Frame 20: __libc_start_main
Frame 21: _start
=====

Received signal 15: Terminated

** StackTrace of 11 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: _dl_addr
Frame 2: LvArray::system::stackTrace[abi:cxx11](bool)
Frame 3: geosx::SpatialPartition::setSizes(double const (&) [3], double const (&) [3])
Frame 4: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 5: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 6: geosx::ProblemManager::generateMesh()
Frame 7: geosx::ProblemManager::problemSetup()
Frame 8: geosx::GeosxState::initializeDataRepository()
Frame 9: main
Frame 10: __libc_start_main
Frame 11: _start
=====

Received signal 15: Terminated

** StackTrace of 16 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: conduit::Schema::object_hierarchy() const
Frame 2: conduit::Schema::object_map[abi:cxx11]() const
Frame 3: conduit::Schema::has_child(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
Frame 4: conduit::Node::fetch(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
Frame 5: geosx::dataRepository::WrapperBase::WrapperBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group&)
Frame 6: geosx::ObjectManagerBase::ObjectManagerBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 7: geosx::ElementSubRegionBase::ElementSubRegionBase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 8: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 9: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 10: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 11: geosx::ProblemManager::generateMesh()
Frame 12: geosx::ProblemManager::problemSetup()
Frame 13: geosx::GeosxState::initializeDataRepository()
Frame 14: main
Frame 15: __libc_start_main
Frame 16: _start
=====

Received signal 15: Terminated

** StackTrace of 21 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: /lib/x86_64-linux-gnu/libstdc++.so.6
Frame 2: std::locale::locale()
Frame 3: std::ios_base::_M_init()
Frame 4: std::basic_ios<char, std::char_traits<char> >::init(std::basic_streambuf<char, std::char_traits<char> >*)
Frame 5: conduit::Schema::name[abi:cxx11]() const
Frame 6: conduit::Schema::path[abi:cxx11]() const
Frame 7: conduit::Schema::path[abi:cxx11]() const
Frame 8: conduit::Schema::path[abi:cxx11]() const
Frame 9: conduit::Schema::path[abi:cxx11]() const
Frame 10: conduit::Schema::path[abi:cxx11]() const
Frame 11: conduit::Node::path[abi:cxx11]() const
Frame 12: geosx::dataRepository::Wrapper<geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> > >& geosx::dataRepository::Group::registerWrapper<geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::InterObjectRelation<LvArray::Array<long, 2, camp::int_seq<long, 0l, 1l>, long, LvArray::ChaiBuffer> >*)
Frame 13: geosx::CellBlock::CellBlock(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, geosx::dataRepository::Group*)
Frame 14: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 15: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 16: geosx::ProblemManager::generateMesh()
Frame 17: geosx::ProblemManager::problemSetup()
Frame 18: geosx::GeosxState::initializeDataRepository()
Frame 19: main
Frame 20: __libc_start_main
Frame 21: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/mpiCommunications/SpatialPartition.cpp:204
***** Controlling expression (should be false): check != m_size
***** Rank 9:
Expected check == m_size
  check = 1
  m_size = 10


** StackTrace of 7 frames **
Frame 0: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 1: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 2: geosx::ProblemManager::generateMesh()
Frame 3: geosx::ProblemManager::problemSetup()
Frame 4: geosx::GeosxState::initializeDataRepository()
Frame 5: main
Frame 6: __libc_start_main
Frame 7: _start
=====

Received signal 15: Terminated

** StackTrace of 15 frames **
Frame 0: /lib/x86_64-linux-gnu/libc.so.6
Frame 1: clock_nanosleep
Frame 2: nanosleep
Frame 3: orte_show_help_norender
Frame 4: orte_show_help
Frame 5: MPI_Abort
Frame 6: geosx
Frame 7: geosx::SpatialPartition::setSizes(double const (&) [3], double const (&) [3])
Frame 8: geosx::InternalMeshGenerator::generateMesh(geosx::DomainPartition&)
Frame 9: geosx::MeshManager::generateMeshes(geosx::DomainPartition&)
Frame 10: geosx::ProblemManager::generateMesh()
Frame 11: geosx::ProblemManager::problemSetup()
Frame 12: geosx::GeosxState::initializeDataRepository()
Frame 13: main
Frame 14: __libc_start_main
Frame 15: _start
=====

[ubuntu:03942] 10 more processes have sent help message help-mpi-api.txt / mpi-abort
[ubuntu:03942] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

````

**To Reproduce**
run test dataset with 10 processors
````
mpirun -np 10 geosx -i CO2.xml
````

**Expected behavior**

**Screenshots**

**Platform (please complete the following information):**
Machine: Ubuntu
Compiler: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
GEOSX Version:0.2.0

any workaround to make this work?

Regards,

Deming

**Comment by francoishamon** [comment]
Hi Deming,
If the mesh is specified with the `InternalMeshGenerator`, you need to specify the number of partitions in the x, y, and z directions. So, assuming that your mesh has enough cells (which is the case in your input file), you can try:
```
mpirun -np 10 geosx -i CO2.xml -x 5 -y 2 -z 1
```
If at some point you switch to `PAMELAMeshGenerator`, the number of partitions in each direction will no longer be needed.

**Comment by wdmhouston** [comment]
Francois,

    Thanks, that fixed the partition issue. Now a new issue about perforation shows up:
```
user@ubuntu:/data/final1_rate$ mpirun -np 9 geosx -i CO2.xml -x 3 -y 3 -z 1
Adding Solver of type CompositionalMultiphaseReservoir, named coupledFlowAndWells
Adding Solver of type CompositionalMultiphaseFVM, named compositionalMultiphaseFlow
Adding Solver of type CompositionalMultiphaseWell, named compositionalMultiphaseWell
Adding Mesh: InternalMesh, mesh
Adding Mesh: InternalWell, wellInjector1
Adding Event: PeriodicEvent, outputs
Adding Event: PeriodicEvent, restarts
Adding Event: PeriodicEvent, timeHistoryCollection
Adding Event: PeriodicEvent, timeHistoryOutput
Adding Event: PeriodicEvent, wellRateCollection
Adding Event: PeriodicEvent, timeRateHistoryOutput
Adding Event: PeriodicEvent, solverApplications
Adding Output: VTK, vtkOutput
Adding Output: Restart, restartOutput
Adding Output: TimeHistory, timeHistoryOutput
Adding Output: TimeHistory, timeRateHistoryOutput
   TableFunction: permxFunc
   TableFunction: permyFunc
   TableFunction: permzFunc
Adding Object CellElementRegion named Reservoir from ObjectManager::Catalog.
Adding Object WellElementRegion named wellRegion from ObjectManager::Catalog.
   TableFunction: CO2DensityTable
   TableFunction: CO2ViscosityTable
   TableFunction: brineDensityTable
   TableFunction: CO2SolubilityTable
***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 3: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 4: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 5: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 6: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 0: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 2: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 7: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 8: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 4 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

***** ERROR
***** LOCATION: /apps/GEOSX/src/coreComponents/mesh/WellElementRegion.cpp:67
***** Controlling expression (should be false): matchedPerforations != numPerforationsGlobal
***** Rank 1: Invalid mapping perforation-to-element in well wellRegion. This happens when GEOSX cannot match a perforation with a reservoir element. The most common reason for this error is that a perforation is on a section of  the well polyline located outside the domain.

** StackTrace of 6 frames **
Frame 0: geosx::ElementRegionManager::generateWells(geosx::MeshManager&, geosx::MeshLevel&)
Frame 1: geosx::ProblemManager::generateMesh()
Frame 2: geosx::ProblemManager::problemSetup()
Frame 3: geosx::GeosxState::initializeDataRepository()
Frame 4: main
Frame 5: __libc_start_main
Frame 6: _start
=====

[ubuntu:01233] 8 more processes have sent help message help-mpi-api.txt / mpi-abort
[ubuntu:01233] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

```
I tried different partition options,  but still no go.  Any idea or workaround?


Thanks,
Deming

**Comment by francoishamon** [comment]
Have you been able to make this work? 

The problem comes from the fact that the perforation is located exactly on the face between two faces, at { 4983.49, 4983.49, 50, 32.6288 }, which GEOSX from matching the perforation to a reservoir cell. If you move slightly the perforation along the well, for instance with:
```
<InternalWell
      name="wellInjector1"
      wellRegionName="wellRegion"
      wellControlsName="wellControls"
      meshName="mesh"
      polylineNodeCoords="{ { 4983.49, 4983.49, 50},
                            { 4983.49, 4983.49, 0} }"
      polylineSegmentConn="{ { 0, 1 } }"
      radius="0.09525"
      numElementsPerSegment="1">
      <Perforation
        name="injector1_perf1"
        distanceFromHead="17.37"/>
    </InternalWell>
```
then everything should work, in serial and in parallel. Can you confirm that this is the case?

**Comment by wdmhouston** [comment]
Francois,
   
    You're right. I moved well trajectory to the middle of cells, and now parallel mode works!   There are hdf5 bugs in parallel mode, and I have already reported in an existing thread(https://github.com/GEOSX/GEOSX/issues/1427). Hope that hdf5 is fixed soon.

    One more question, if I want to make a horizontal well, I can just update polylineNodeCoords to make well trajectory horizontal, for example,
````
polylineNodeCoords="{ { 5025, 4525, 25},
                            { 5025, 5525, 25} }"
````
Is this correct? If no, please advise.

**Comment by francoishamon** [comment]
This issue seems to have been resolved so I am closing this.
